From e3a1593a0984cc0156ec1892a46af8f256a64878 Mon Sep 17 00:00:00 2001
From: Daniel Boles <dboles.src@gmail.com>
Date: Thu, 13 Dec 2018 17:20:13 +0100
Subject: [PATCH] x11: Fix deprecation macro use

G_GNUC_END_IGNORE_DEPRECATIONS terminates the if statement and does not
consider the following block to be part of the if. So that block was
always taken irregardless of the pattern.

Fixes #1280
---
 gdk/x11/gdkwindow-x11.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gdk/x11/gdkwindow-x11.c b/gdk/x11/gdkwindow-x11.c
index 97ada6d739..764e39495f 100644
--- a/gdk/x11/gdkwindow-x11.c
+++ b/gdk/x11/gdkwindow-x11.c
@@ -2985,6 +2985,7 @@ gdk_window_x11_set_background (GdkWindow      *window,
   double r, g, b, a;
   cairo_surface_t *surface;
   cairo_matrix_t matrix;
+  cairo_pattern_t *parent_relative_pattern;
 
   if (GDK_WINDOW_DESTROYED (window))
     return;
@@ -2997,8 +2998,10 @@ gdk_window_x11_set_background (GdkWindow      *window,
     }
 
 G_GNUC_BEGIN_IGNORE_DEPRECATIONS
-  if (pattern == gdk_x11_get_parent_relative_pattern ())
+  parent_relative_pattern = gdk_x11_get_parent_relative_pattern ();
 G_GNUC_END_IGNORE_DEPRECATIONS
+
+  if (pattern == parent_relative_pattern)
     {
       GdkWindow *parent;
 
-- 
2.18.1

